src/test/java/com/hotels/bdp/circustrain/bigquery/table/service/partitioned/PartitionedTableServiceTest.java

```diff
@@ -74,7 +75,8 @@ public void getTable() {
   public void getPartitions() {
     when(tableResult.iterateAll()).thenReturn(rows);
     List<Partition> result = partitionedTableService.getPartitions();
-    assertThat(result, is(factory.generate(eq(partitionBy), eq(rows))));
+    System.out.println(result);
+    assertThat(result, is(partitionGenerator.generate(eq(partitionBy), eq(type), eq(rows))));
```

Comment:
like this:
```    List<Partition> expected = Lists.newArrayList(partition);
    when(partitionGenerator.generate(eq(partitionBy), eq(type), eq(rows))).thenReturn(expected);
    assertThat(result, is(expected));
```
You have to mock the `generate` method otherwise it makes no sense, you'll always get the default mocking behaviour back which is return null no matter what the arguments are.